contracts/access/README.adoc

```diff
@@ -5,8 +5,94 @@ NOTE: This document is better viewed at https://docs.openzeppelin.com/contracts/
+** *Done:* An operation that has been executed.
+* *Predecessor*: An (optional) dependency between operations. An operation can depend on another operation (its predecessor), forcing the execution order of these two operations.
+* *Role*:
+** *Proposer:* An address (smart contract or EOA) that is in charge of scheduling (and cancelling) operations.
+** *executor:* An address (smart contract or EOA) that is in charge of executing operations.
+
+=== Operation structure
+
+Operation executed by the xref:api:access.adoc#TimelockController[`TimelockControler`] can contain one or multiple subsequent calls. Depending on whether you need to multiple calls to be executed atomically, you can either use simple or batched operations.
+
+==== Simple operations contain:
+
+* A *target*, the address of the smart contract that the timelock should operate on.
+* A *value*, in wei, that should be sent with the transaction. Most of the time this will be 0. Ether can be deposited before-end or passed along when executing the transaction.
+* A *data* field, containing the encoded function selector and parameters of the call. This can be produced using a number of tools. For example, a maintenance operation granting role `ROLE` to `ACCOUNT` can be encode using web3js as follows:
+
+```javascript
+const data = timelock.contract.methods.grantRole(ROLE, ACCOUNT).encodeABI()
+```
+
+* A *predecessor*, that specifies a dependency between operations. This dependency is optional. Use `bytes32(0)` if the operation does not have any dependency.
+* A *salt*, used to disambiguate two otherwise identical operations. This can be any random value.
+
+==== Batched operations contain:
+
+* An array (*targets*) of targets.
+* An array (*values*) of values.
+* An array (*datas*) of data fields.
+* A *predecessor*, that specifies a dependency between operations. This dependency is optional. Use `bytes32(0)` if the operation does not have any dependency.
+* A *salt*, used to disambiguate two otherwise identical operations. This can be any random value.
+
+TIP: The three arrays must have the same length. They contain the details of the calls, just like in a simple operation.
+
+=== Operation lifecycle
+
+Timelocked operations are identified by a unique id (their hash) and follow a specific lifecycle:
+
+`Unset` -> `Pending` -> `Pending` + `Ready` -> `Done`
+
+* By calling xref:api:access.adoc#TimelockController-schedule-address-uint256-bytes-bytes32-bytes32-uint256-[`schedule`] (or xref:api:access.adoc#TimelockController-scheduleBatch-address---uint256---bytes---bytes32-bytes32-uint256-[`scheduleBatch`]), a proposer moves the operation from the `Unset` to the `Pending` state. This starts a timer that must be longer than the minimum delay. The timer expires at a timestamp accessible through the xref:api:access.adoc#TimelockController-getTimestamp-bytes32-[`getTimestamp`] methods.
+* Once the timer expires, the operation gets the `Ready` state. At this point, it can be executed.
```

Comment:
May be worth being explicit that this transition is automatic:

```suggestion
* Once the timer expires, the operation automatically gets the `Ready` state. At this point, it can be executed.
```